### PR TITLE
minor changes

### PR DESCRIPTION
**Pull request by mohan08p**

### CODE REVIEW

File: developer-tools/java/chapters/ch03-build-image.adoc [hunk 1]
```diff
@@ -10,11 +10,12 @@ As explained in <<Docker_Basics>>, Docker image is the *build component* of Dock
 
 Docker build images by reading instructions from a _Dockerfile_. A _Dockerfile_ is a text document that contains all the commands a user could call on the command line to assemble an image. `docker build` command uses this file and executes all the commands in succession to create an image.
 
-`build` command is also passed a context that is used during image creation. This context can be a path on your local filesystem or a URL to a Git repository.
+`build` command is also passed a context that is used during image creation. This context can be a path to your local filesystem or a URL to a Git repository.
```
Comment (ManoMarks) [inline developer-tools/java/chapters/ch03-build-image.adoc]: "on is correct in this instance"

File: developer-tools/java/chapters/ch03-build-image.adoc [hunk 2]
```diff
@@ -10,11 +10,12 @@ As explained in <<Docker_Basics>>, Docker image is the *build component* of Dock
...
 
-_Dockerfile_ is usually called _Dockerfile_. The complete list of commands that can be specified in this file are explained at https://docs.docker.com/reference/builder/. The common commands are listed below:
+_Dockerfile_ is usually called _Automate Building of Images_. The complete list of commands that can be specified in this file are explained at https://docs.docker.com/reference/builder/. The common commands are listed below:
```
Comment (ManoMarks) [inline developer-tools/java/chapters/ch03-build-image.adoc]: Default name for a Dockerfile is "Dockerfile" and that convention is what is usually used

File: developer-tools/java/chapters/ch02-basic-concepts.adoc [hunk 3]
```diff
@@ -14,7 +14,7 @@ Docker simplifies software delivery by making it easy to build and share images
 
 Your application typically require a specific version of operating system, application server, JDK, database server, may require to tune the configuration files, and similarly multiple other dependencies. The application may need binding to specific ports and certain amount of memory. The components and configuration together required to run your application is what is referred to as application operating system.
 
-You can certainly provide an installation script that will download and install these components. Docker simplifies this process by allowing to create an image that contains your application and infrastructure together, managed as one component. These images are then used to create Docker containers which run on the container virtualization platform, provided by Docker.
+You can certainly provide an installation script that will download and install these components. Docker simplifies this process by allowing us to create an image that contains your application and infrastructure together, managed as one component. These images are then used to create Docker containers which run on the container virtualization platform, provided by Docker.
```
Comment (ManoMarks) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: I agree we should have a pronoun, but let's use "you" to be consistent.

File: developer-tools/java/chapters/ch02-basic-concepts.adoc [hunk 4]
```diff
@@ -24,22 +24,23 @@ Docker has three main components:
 . __Containers__ are the *run component* of Docker and created from images. Containers can be run, started, stopped, moved, and deleted.
 . Images are stored, shared, and managed in a __registry__ and are the *distribution component* of Docker. Docker Hub is a publicly available registry and is available at http://hub.docker.com.
 
-In order for these three components to work together, the *Docker Daemon* (or Docker Engine) runs on a host machine and does the heavy lifting of building, running, and distributing Docker containers. In addition, the *Client* is a Docker binary which accepts commands from the user and communicates back and forth with the Engine.
+In order to work these three components together, the *Docker Daemon* (or Docker Engine) runs on a host machine and does the heavy lifting of building, running, and distributing Docker containers. In addition, the *Client* is a Docker binary which accepts commands from the user and communicates back and forth with the Engine.
```
Comment (ManoMarks) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: No, this changes the meaning. The components want to work together. If you "work these three components together" you mean you're combining them, which isn't the case.

File: developer-tools/java/chapters/ch02-basic-concepts.adoc [hunk 5]
```diff
@@ -24,22 +24,23 @@ Docker has three main components:
...
+One of the reason Docker is so lightweight is because of these layers. When you change a Docker image—for example, updating an application to a new version— a new layer gets built and added on top of it. Thus, rather than replacing the whole image or entirely rebuilding, as you may do with a virtual machine, only that layer is added or updated. Now you don't need to distribute a whole new image, just the update, making distribution of Docker images faster and simpler.
 
-Every image starts from a base image, for example `ubuntu`, a base Ubuntu image, or `fedora`, a base Fedora image. You can also use images of your own as the basis for a new image, for example if you have a base Apache image you could use this as the base of all your web application images.
+Every image starts from a base image, for example `ubuntu`, a base Ubuntu image, or `fedora`, a base Fedora image. You can also use images of your own as the basis or foundation for a new image, for example if you have a base Apache image you could use this as the base of all your web application images.
```
Comment (ManoMarks) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: This is repetitive. "basis" means "foundation" in this case.

File: developer-tools/java/chapters/ch02-basic-concepts.adoc [hunk 6]
```diff
@@ -24,22 +24,23 @@ Docker has three main components:
...
 
-One of the reasons Docker is so lightweight is because of these layers. When you change a Docker image—for example, update an application to a new version— a new layer gets built. Thus, rather than replacing the whole image or entirely rebuilding, as you may do with a virtual machine, only that layer is added or updated. Now you don't need to distribute a whole new image, just the update, making distributing Docker images faster and simpler.
+One of the reason Docker is so lightweight is because of these layers. When you change a Docker image—for example, updating an application to a new version— a new layer gets built and added on top of it. Thus, rather than replacing the whole image or entirely rebuilding, as you may do with a virtual machine, only that layer is added or updated. Now you don't need to distribute a whole new image, just the update, making distribution of Docker images faster and simpler.
```
Comment (hairyhenderson) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: Should leave `reasons` alone. Reasons is plural in this context.

Comment (hairyhenderson) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: `distributing` was less stilted than `distribution of` in this context IMO

File: developer-tools/java/chapters/ch02-basic-concepts.adoc [hunk 7]
```diff
@@ -48,11 +49,11 @@ Docker images are then built from these base images using a simple, descriptive
...
 
 **How does a Container work?**
 
-A container consists of an operating system, user-added files, and meta-data. As we've seen, each container is built from an image. That image tells Docker what the container holds, what process to run when the container is launched, and a variety of other configuration data. The Docker image is read-only. When Docker runs a container from an image, it adds a read-write layer on top of the image (using a union file system as we saw earlier) in which your application can then run.
+A container consists of an operating system, user-added files, and meta-data. As we've seen, each container is built from an image. That image tells Docker what the container holds, what process to run when the container is launched, and a variety of other configuration data. The Docker images are read-only templates. When Docker runs a container from an image, it adds a read-write layer on top of the image (using a union file system as we saw earlier) in which your application can then run.
```
Comment (hairyhenderson) [inline developer-tools/java/chapters/ch02-basic-concepts.adoc]: `images are` is inconsistent - the previous sentences are discussing the singular concept of an image. Also I'm not sure what `templates` means in this context - a Docker image isn't a template.

**Comment by ManoMarks** [comment]
Actually as all the changes but one have a negative response, I'm closing this PR.

**Comment by mohan08p** [comment]
@ManoMarks @hairyhenderson : sorry.. even I'll stash into my local repo...

**Comment by arun-gupta** [comment]
Thanks @ManoMarks 

@hairyhenderson keep us honest, feel free to send an updated PR with the change that matters to you and would make the wordings clear.

**Pull request closed without merging**